gcm.go

```diff
@@ -194,6 +195,12 @@ func (c *httpGcmClient) send(apiKey string, m HttpMessage) (*HttpResponse, error
 		return nil, fmt.Errorf("error sending request to HTTP connection server>%v", err)
 	}
 	gcmResp := &HttpResponse{}
+	gcmResp.Status = httpResp.StatusCode
+
+	if gcmResp.Status != http.StatusOK {
+		return gcmResp, fmt.Errorf("Encountered HTTP status code %d", gcmResp.Status)
```

Comment:
@jpatel531  You are ditching the response body, which can contain useful info. A better way on my opinion is to always read the body, try to unmarshal it, and if failed return the body as is.